registry/auth.go

```diff
@@ -271,14 +269,10 @@ func Login(authConfig *AuthConfig, factory *utils.HTTPRequestFactory) (string, e
 }
 
 // this method matches a auth configuration to a server address or a url
-func (config *ConfigFile) ResolveAuthConfig(hostname string) AuthConfig {
-	if hostname == IndexServerAddress() || len(hostname) == 0 {
-		// default to the index server
-		return config.Configs[IndexServerAddress()]
-	}
-
+func (config *ConfigFile) ResolveAuthConfig(index *IndexInfo) AuthConfig {
+	configKey := index.GetAuthConfigKey()
 	// First try the happy case
-	if c, found := config.Configs[hostname]; found {
+	if c, found := config.Configs[configKey]; found || index.Official {
```

Comment:
That is what the old code did.  The removed block above (if hostname == Index...) would return whatever config.Config had for the official index, even if it was empty.